src/tool/xmeta/cpp/ast_to_st_listener.cpp

```diff
@@ -320,36 +327,44 @@ void ast_to_st_listener::exitEnum_declaration(XlangParser::Enum_declarationConte
             return;
         }
     }
-
-    m_reader.m_cur_namespace_body->add_enum(new_enum);
-    m_reader.symbols[symbol] = new_enum;
+    if (!xlang_model.set_symbol(symbol, new_enum))
```

Comment:
it seems a little unintuitive that the code here can infer from a Boolean result the specific nature of the error occurred.

Reply:
Fixing in later PR.